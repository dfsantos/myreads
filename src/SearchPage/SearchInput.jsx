import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  onChange: PropTypes.func.isRequired,
};

function SearchInput({ onChange }) {
  return (
    <div className="search-books-input-wrapper">
      <input type="text" placeholder="Search by title or author" onChange={onChange} />
    </div>
  );
}

SearchInput.propTypes = propTypes;

export default SearchInput;
