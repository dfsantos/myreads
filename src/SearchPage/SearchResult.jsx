import React from 'react';
import PropTypes from 'prop-types';

import Book from '../common/Book';

const propTypes = {
  books: PropTypes.array,
};

const defaultProps = {
  books: [],
};

const toBookItem = book => (
  <li key={book.id}>
    <Book title={book.title} />
  </li>
);

function SearchResult({ books }) {
  return (
    <div className="search-books-results">
      <ol className="books-grid">{books.map(toBookItem)}</ol>
    </div>
  );
}

SearchResult.propTypes = propTypes;
SearchResult.defaultProps = defaultProps;

export default SearchResult;
