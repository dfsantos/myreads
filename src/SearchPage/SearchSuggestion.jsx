import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'underscore';

const propTypes = {
  words: PropTypes.arrayOf(PropTypes.string).isRequired,
  pause: PropTypes.bool,
};

const defaultProps = {
  pause: false,
};

class SearchSuggestion extends Component {
  constructor(props) {
    super(props);
    this.state = { words: _.shuffle(props.words) };
    this.suggestWord = this.suggestWord.bind(this);
    this.stop = setInterval(this.suggestWord, 2500);
  }

  componentWillUnmount() {
    clearInterval(this.stop);
  }

  suggestWord() {
    this.setState(state => {
      const { words } = state;
      words.unshift(words.pop());
      return { words };
    });
  }

  render() {
    return (
      <div>
        {!this.props.pause && (
          <div className="search-suggestion">
            Try search for <br />
            "{this.state.words[0]}"
          </div>
        )}
      </div>
    );
  }
}

SearchSuggestion.propTypes = propTypes;
SearchSuggestion.defaultProps = defaultProps;

export default SearchSuggestion;
