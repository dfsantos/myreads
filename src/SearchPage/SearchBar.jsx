import React from 'react';
import PropTypes from 'prop-types';
import SearchInput from './SearchInput';

const propTypes = {
  onSearch: PropTypes.func.isRequired,
};

function SearchBar({ onSearch }) {
  return (
    <div className="search-books-bar">
      <a href="/" className="close-search">
        Close
      </a>
      <SearchInput onChange={onSearch} />
    </div>
  );
}

SearchBar.propTypes = propTypes;

export default SearchBar;
