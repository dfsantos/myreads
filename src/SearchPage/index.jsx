import React, { Component } from 'react';
import PropTypes from 'prop-types';

import SearchBar from './SearchBar';
import SearchResult from './SearchResult';

import * as BooksAPI from '../BooksAPI';

const EVENT_WRAPPER = { target: { value: '' } };

const propTypes = {
  onCategorizeBook: PropTypes.func.isRequired,
};

class SearchPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      searchResult: [],
    };

    this.onSearch = this.onSearch.bind(this);
    this.onCategorizeBook = this.onCategorizeBook.bind(this);
  }

  async onSearch(event = EVENT_WRAPPER) {
    const searchQuery = event.target.value;
    const searchResult = await BooksAPI.search(searchQuery, 10);
    this.setState({ searchResult });
  }

  onCategorizeBook(book) {
    return category => this.props.onCategorizeBook(Object.assign(book, { category }));
  }

  render() {
    const { searchResult } = this.state;
    const { onCategorizeBook } = this.props;
    const { onSearch } = this;
    return (
      <div className="search-books">
        <SearchBar onSearch={onSearch} />
        <SearchResult books={searchResult} onCategorizeBook={onCategorizeBook} />
      </div>
    );
  }
}

SearchPage.propTypes = propTypes;

export default SearchPage;
