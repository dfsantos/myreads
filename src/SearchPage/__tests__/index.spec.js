import React from 'react';
import toJson from 'enzyme-to-json';

import SearchPage from '../index';

it('should render without errors', () => {
  shallow(<SearchPage />);
});

it('default snapshot', () => {
  const wrapper = shallow(<SearchPage />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
