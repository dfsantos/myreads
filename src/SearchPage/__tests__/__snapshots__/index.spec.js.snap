// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default snapshot 1`] = `
<div
  className="search-books"
>
  <SearchBar
    onSearch={[Function]}
  />
  <SearchResult
    books={Array []}
  />
</div>
`;
